scripts/benchmark_ad.py

```diff
@@ -0,0 +1,130 @@
+import os
+import argparse
+from typing import List
+import tempfile
+import pytorch_lightning as pl
+from guided_diffusion import logger
+from utils.models import ModelLoader
+from utils.noiser import TimestepUniformNoiser
+from utils.denoiser import ModelTimestepUniformDenoiser
+from utils.error_map import BatchFilteredSquaredError
+from utils.anomaly_scorer import MaxValueAnomalyScorer
+from core.diffusion_ad import DiffusionAD
+from config.configuration import CATEGORY_TO_TYPE, DIFFUSION_AD_HPARAMS, DEFAULT_ROOT_OUTPUT_DIR
+
+
+DEFAULT_LOG_BASE = DEFAULT_ROOT_OUTPUT_DIR
+os.makedirs(DEFAULT_LOG_BASE, exist_ok=True)
+tempfile.tempdir = DEFAULT_LOG_BASE
```

Comment:
so first you create the path and then you use the tempdir?
what is tempdir and how you use it in this context?
why not in main?
or at least in a function called:
`def init_environ():`
called from `main()`?
it looks weird to have in the main script an assignment from a global to a global.

Reply:
I want to be 1000% sure that this file changes this variable before anything has a chance to access this module.
This will make sure that our files get outputted where we want them to be

Reply:
weird excuse but ok i guess?